service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/s3storageFolder/ControlledAwsS3StorageFolderHandler.java

```diff
@@ -29,9 +31,36 @@ public WsmResource makeResourceFromDb(DbResource dbResource) {
         dbResource, attributes.getBucketName(), attributes.getPrefix());
   }
 
+  /**
+   * Generate controlled AWS S3 Storage folder cloud name that meets the requirements for a valid
+   * name.
+   *
+   * <p>Alphanumeric characters and certain special characters can be safely used in valid names For
+   * details, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-keys.html
+   */
   @Override
-  public String generateCloudName(@Nullable UUID workspaceUuid, String resourceName) {
-    // TODO(TERRA-504): Implement generateCloudName
-    throw new FeatureNotSupportedException("Generate cloud name feature is not implemented yet");
+  public String generateCloudName(@Nullable UUID workspaceUuid, String storageFolderName) {
```

Comment:
Could this be static?  Also seems like a good unit test candidate.

Reply:
This function (and makeResourceFromDb) are defined in the interface WsmResourceHandler. making the generateCloudName static requires adding a default definition in the interface. which in turn lets the implementing classes avoid implementing the function. 
At the very least the error message to throw in implementing classes are different (not implemented message, but the resource type in the error string is different).
in addition, the GCP implementation makes use of gcpCloudContextService, which demands that the function is not static. 

hence retaining this as non-static

Reply:
ut in https://verily.atlassian.net/browse/TERRA-470

Reply:
+1 to unit testing this function